File: NWD.f90
!!! https://pl.spoj.com/problems/PP0501A/

program TEST
    implicit none
    integer :: p, q, n, i

    read(*, *) n
    do i = 1, n
        read(*, *) p, q
        if (p == 0) then
            print *, q
        else if (q == 0) then
            print *, p
        else
            do while (p /= q)
                if (q > p) then
                    q = q - p
                else
                    p = p - q
                end if
            end do
            print *, p
        end if
    end do
end program TEST

File: NWD.swift
#!/usr/bin/env xcrun swift
/// https://pl.spoj.com/problems/PP0501A/

func nwd(_ m: Int, _ n: Int) -> Int {
    var (a, b) = (m, n)
    while b != 0 {
        (a, b) = (b, a % b)
    }
    return a
}

for _ in 1...Int(readLine()!)! {
    let t = readLine()!
        .split(separator: " ")
        .map { Int($0)! }
    print(nwd(t[0], t[1]))
}
